test/common/http/conn_pool_grid_test.cc

```diff
@@ -155,8 +210,35 @@ TEST_F(ConnectivityGridTest, FailureThenSuccessForMultipleConnections) {
   cancel2->cancel(Envoy::ConnectionPool::CancelPolicy::CloseExcess);
 }
 
-// Test both pools failing
-TEST_F(ConnectivityGridTest, FailureTwice) {
+// Test both connections happening in parallel and both failing.
+TEST_F(ConnectivityGridTest, DoubleFailureParallel) {
```

Comment:
Should we also add a separate DoubleFailureSerial test?

Reply:
IMO not worth it - the serial code is pretty trivial compared to parallel but I can add in a follow-up if you think it's worth having

Reply:
Fine by me!